test/MSVC/query_vcbat.py

```diff
@@ -51,7 +51,7 @@
 #            print k, v
 #MergeMSVSBatFile(env, 9.0)
 #print(env['ENV']['PATH'])
-print(query_versions())
+print(query_versions(env=None))
```

Comment:
Doesn't the env default to none here? Better to test without specifying then I think users won't typically specify env=None..

Reply:
> Doesn't the env default to none here?

It does *now*, it did not take any arguments before.

I'm happy to remove if you think it best.  But...

This is an artifact of one of issues with the user specification of the VSWHERE construction variable.

When the VSWHERE construction variable was implemented, not all of the call chain "roots" where the VSWHERE construction variable *could* be populated were addressed.  This was not obvious due to the environment argument being optional in many function implementations.  Basically, all of the call chains needed to accept an optional environment argument if one was not already defined.

Mini-rant: the optional environment argument makes the vc code a lot harder to comprehend because it is not readily apparent what might be retrieved from the environment, if it exists, in the bowels of the call chains.

The function `query_versions` did not accept any arguments *but* could cause resolution of the vswhere location in the vc code.  Because `query_versions` did not accept an argument, it was not possible to pass an environment that might contain the construction variable VSWHERE.

The `query_versions` function call in the test was changed to take an argument explicitly assigning None to the environment as a *visual reminder* that an optional environment argument is now accepted.  When doing source tree diffs and searches, it was easier to determine if all of the call chains were inspected.

In the future, a test could be added that accepted a VSWHERE executable location which *might* be something that a user would want to specify.